lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -497,11 +531,14 @@ Value FIRRTLTypesLowering::addArg(Type type, unsigned oldArgNumber,
   // Append the new argument.
   auto newValue = body->addArgument(type);
 
-  // Copy over the name attribute for the new argument.
+  // Save the name attribute for the new argument.
   StringAttr nameAttr = getFIRRTLNameAttr(module.getArgAttrs(oldArgNumber));
   if (nameAttr) {
     auto newArgAttrs = getArgAttrs(nameAttr, nameSuffix, builder);
-    module.setArgAttrs(newValue.getArgNumber(), newArgAttrs);
+    auto newArgNumber = newValue.getArgNumber() - originalNumArgs;
+    auto newArgAttrName = getArgAttrName(newArgNumber);
+    auto newAttr = builder->getNamedAttr(newArgAttrName, newArgAttrs);
+    newAttrs.push_back(newAttr);
```

Comment:
Once getArgAttrName returns an identifier, you can just use:

`newAttrs.push_back(NamedAttribute(newArgAttrName, newArgAttrs));`

or:

`newAttrs.push_back({newArgAttrName, newArgAttrs});`

instead of calling getNamedAttr